test/test_id_model.cpp

```diff
@@ -57,14 +57,34 @@ Expr* getParentExpr(Val* val, int n) {
   return val->definition();
 };
 
-TensorView* getTensorByName(
-    const std::vector<TensorView*>& tvs,
-    StmtNameType name) {
+IterDomain* getParentId(IterDomain* id, int n) {
+  for (int i = 0; i < n; ++i) {
+    NVF_ERROR(id->definition() != nullptr);
+    NVF_ERROR(id->definition()->input(0)->isA<IterDomain>());
+    id = id->definition()->input(0)->as<IterDomain>();
+  }
+  NVF_ERROR(id != nullptr);
+  return id;
+};
+
+// Get the n-th descendant by traversing a sibling
+IterDomain* getChildId(IterDomain* id, int n, int sibling_idx = 0) {
+  for (int i = 0; i < n; ++i) {
+    NVF_ERROR(!id->uses().empty());
+    NVF_ERROR(id->uses().front()->output(sibling_idx)->isA<IterDomain>());
+    id = id->uses().front()->output(sibling_idx)->as<IterDomain>();
+  }
+  NVF_ERROR(id != nullptr);
+  return id;
+};
+
+template <typename ValType>
+ValType* getValByName(const std::vector<ValType*>& vals, StmtNameType name) {
```

Comment:
I am wondering how stable the val names will be as we evolve our system. Will this become as fragile as the tests that uses
```C++
// assert that the given fusion lowers to the given CUDA kernel
void assertCUDAKernel(Fusion* fusion, const std::string& expected_kernel);
```
and we need to update this test again and again?

Reply:
Generally yes, but for these tests, I'm just using this function only for the tensor views originally defined in the initial fusion, so scheduling or ID replays won't affect their names. So, I believe they should be stable.